Refine file naming scheme for profiling data

ChinYikMing opened the pull request:
Close #336

jserv commented on src/main.c:
```diff
@@ -167,10 +169,10 @@ static bool parse_args(int argc, char **args)
+        char *prog_basename = basename(opt_prog_name);
+        prof_out_file = malloc(strlen(prog_basename) + 12 + 1);
+        assert(prof_out_file);
+        sprintf(prof_out_file, "./prof/%s.prof", prog_basename);
```
The leading `prof/` directory name is not really making sense. Instead, let's place the profiling data in the same directory where the RV32 ELF file locates.

ChinYikMing replied:
I think you mean `build` directory

qwe661234 replied:
I think putting it in the `build` directory is reasonable, we can distinguish the executable files and profiling data by filename extension.

ChinYikMing commented:
Note that the trailing spaces are removed automatically by editor when I motify "README" and "rv_profiler", not my primary intention.

jserv commented on src/main.c:
```diff
@@ -167,10 +169,10 @@ static bool parse_args(int argc, char **args)
+        char *prog_basename = basename(opt_prog_name);
+        prof_out_file = malloc(strlen(prog_basename) + 11 + 1);
+        assert(prof_out_file);
+        sprintf(prof_out_file, "build/%s.prof", prog_basename);
```
Is the leading `build/` necessary?

ChinYikMing replied:
It depends on working directory of `rv32emu` executable. There are 2 cases to run profiler:

1. 
```
// outside build directory
xxx/build/rv32emu -p xxx/build/xxx.elf
```

2. 
```
// inside build directory
./rv32emu -p xxx.elf
```

Without leading `build/`, the first case will always write to cwd which is parent directory of `xxx`. For second case, it is all right.

I think the new approach could be concatenating the cwd and relative path (e.g., `xxx/build/` in first case and `./` in second case) to `rv32emu` executable. This concatenated path will always point to the `build` directory.

jserv replied:
> I think the new approach could be concatenating the cwd and relative path (e.g., `xxx/build/` in first case and `./` in second case) to `rv32emu` executable. This concatenated path will always point to the `build` directory.

I think the new one is better, and I expect a cleaner and elegant implementation.

ChinYikMing committed:
Refine file naming scheme for profiling data

- fix incorrect basename generated by dynamic profiler
- force profiling data store in build directory regardless
  of the cwd
- update README

Close #336

jserv commented:
Thank @ChinYikMing for contributing!
